File: mobile/src/api/diseaseApi.ts
import axios from 'axios';

const API_BASE_URL = process.env.API_BASE_URL || 'http://localhost:8080/api/v1';

export const diseaseApi = {
  /**
   * Upload a leaf image for disease detection.
   * Multipart form-data POST to /api/v1/disease/detect
   */
  async detect(params: {
    file: { uri: string; name: string; type: string };
    farmer_id: string;
    crop_id?: string;
    crop_type: string;
    language?: string;
    authToken: string;
  }) {
    const formData = new FormData();

    formData.append('file', {
      uri: params.file.uri,
      name: params.file.name || 'leaf.jpg',
      type: params.file.type || 'image/jpeg',
    } as any);

    formData.append('farmer_id', params.farmer_id);
    formData.append('crop_type', params.crop_type);
    formData.append('language', params.language || 'marathi');

    if (params.crop_id) {
      formData.append('crop_id', params.crop_id);
    }

    const response = await axios.post(
      `${API_BASE_URL}/disease/detect`,
      formData,
      {
        headers: {
          'Content-Type': 'multipart/form-data',
          Authorization: `Bearer ${params.authToken}`,
        },
      }
    );

    return response.data;
  },
};

File: mobileV2/app/src/main/java/com/krishimitra/mobilev2/data/api/DiseaseApi.kt
package com.krishimitra.mobilev2.data.api

import com.krishimitra.mobilev2.data.model.DiseaseDetectionResponse
import okhttp3.MultipartBody
import okhttp3.RequestBody
import retrofit2.Call
import retrofit2.http.Multipart
import retrofit2.http.POST
import retrofit2.http.Part

/**
 * Retrofit interface for Disease Detection API endpoints.
 * Equivalent to Spring Boot @RestController for disease detection.
 */
interface DiseaseApi {
    
    /**
     * Detect disease from plant image.
     * POST /api/v1/disease/detect
     * Expects multipart/form-data with file, farmer_id, crop_id, crop_type
     */
    @Multipart
    @POST("disease/detect")
    fun detectDisease(
        @Part("farmer_id") farmer_id: RequestBody,
        @Part("crop_id") crop_id: RequestBody?,
        @Part("crop_type") crop_type: RequestBody,
        @Part("language") language: RequestBody,
        @Part file: MultipartBody.Part
    ): Call<DiseaseDetectionResponse>
}